File: 프로그래머스/level1_시저 암호.py
def solution(s, n):
    answer = ''
    for value in s:
        if value == ' ':
            answer += ' '
            continue
        change = ord(value) + n
        if 97 <= ord(value) <= 122:
            if change > 122:
                answer += chr(change - 26)
            else:
                answer += chr(change)
        else:
            if change > 90:
                answer += chr(change - 26)
            else:
                answer += chr(change)

    return answer

print(solution("AB", 1))
print(solution("z", 1))
print(solution("a B z", 4))

# for i in range(len(s)):
#         if s[i].isupper():
#             s[i]=chr((ord(s[i])-ord('A')+ n)%26+ord('A'))
#         elif s[i].islower():
#             s[i]=chr((ord(s[i])-ord('a')+ n)%26+ord('a'))
#
#     return "".join(s)

File: 프로그래머스 복습(java)/src/level1/level1_시저_암호.java
package level1;

public class level1_시저_암호 {
    static class Solution {
        public String solution(String s, int n) {
            String answer = "";
            char[] c = s.toCharArray();
            for (int i = 0; i < s.length(); i++) {
                char alpha = c[i];
                if (alpha == ' ') {
                    answer += ' ';
                } else if (alpha >= 'a' && alpha <= 'z') {
                    if (alpha + n > 'z') {
                        answer += (char) (alpha + n - 26);
                    } else {
                        answer += (char) (alpha + n);
                    }
                } else {
                    if (alpha + n > 'Z') {
                        answer += (char) (alpha + n - 26);
                    } else {
                        answer += (char) (alpha + n);
                    }
                }
            }
            return answer;
        }
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        System.out.println(solution.solution("AB", 1));
        System.out.println(solution.solution("z", 1));
        System.out.println(solution.solution("a B z", 4));
    }
}
